Sources/SwifterSwift/Foundation/DateExtensions.swift

```diff
@@ -257,7 +257,11 @@ public extension Date {
             return calendar.component(.nanosecond, from: self)
         }
         set {
+            #if targetEnvironment(macCatalyst)
+            let allowedRange = 0..<1000000000
```

Comment:
We should probably add a comment here for why this is required. And also why is that value ... is there a way to get this max range dynamically?

Reply:
Comments added. The proper way would be to use the `Calendar` method, but for some reason it returns `nil` in `macCatalyst` as opposed to the proper range. That a nanosecond is 1,000,000,000th of a second I think is safe to hard-code, no?